### PR TITLE
[NCLSUP-198] IGNORE strategy is not used when running 'pig run'

### PR DESCRIPTION
**Pull request by jomrazek**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/bacon/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: pig/src/main/java/org/jboss/pnc/bacon/pig/PigFacade.java [hunk 1]
```diff
@@ -161,29 +165,45 @@ public static GroupBuildInfo run(
         // TODO: should we bother with this case?
 
         RepositoryData repo = null;
-
-        if (repoZipPath != null) {
-            repo = parseRepository(new File(repoZipPath));
+        
+        if (repoZipPath != null || context.getPigConfiguration()
```
Comment (jomrazek) [inline pig/src/main/java/org/jboss/pnc/bacon/pig/PigFacade.java:169]: if strategy is ignore and there is not repoZip specified, we're not generating repo

File: pig/src/main/java/org/jboss/pnc/bacon/pig/PigFacade.java [hunk 2]
```diff
@@ -161,29 +165,45 @@ public static GroupBuildInfo run(
...
+            context.storeContext();
         } else {
-            repo = generateRepo(removeGeneratedM2Dups, configurationDirectory, strictLicenseCheck);
+            log.info("Skipping Repo Generation");
         }
-        context.setRepositoryData(repo);
-        context.storeContext();
 
-        if (!skipSources) {
+        if (!(skipSources || context().getPigConfiguration()
```
Comment (jomrazek) [inline pig/src/main/java/org/jboss/pnc/bacon/pig/PigFacade.java:184]: if skipSources is true or strategy is ignore we're skipping

File: pig/src/main/java/org/jboss/pnc/bacon/pig/PigFacade.java [hunk 3]
```diff
@@ -161,29 +165,45 @@ public static GroupBuildInfo run(
...
+        if (!(skipJavadoc || context.getPigConfiguration()
+                .getFlow()
+                .getJavadocGeneration()
+                .getStrategy() == JavadocGenerationStrategy.IGNORE)) {
             generateJavadoc();
         } else {
             log.info("Skipping Javadoc Generation");
         }
 
-        if (!skipLicenses && repo != null) {
```
Comment (jomrazek) [inline pig/src/main/java/org/jboss/pnc/bacon/pig/PigFacade.java:186]: I changed it as other ones, if user doesn't specify skipLicense and strategy is not IGNORE, it's user error.

File: pig/src/main/java/org/jboss/pnc/bacon/pig/PigFacade.java (hunk 2 above)
Comment (thescouser89) [reply pig/src/main/java/org/jboss/pnc/bacon/pig/PigFacade.java:184]: Should that be `!= IGNORE` ?

File: pig/src/main/java/org/jboss/pnc/bacon/pig/PigFacade.java [hunk 4]
```diff
@@ -162,28 +166,44 @@ public static GroupBuildInfo run(
...
 
         } else {
             log.info("Skipping Source Generation");
         }
 
-        if (!skipJavadoc) {
+        if (!(skipJavadoc || context.getPigConfiguration()
+                .getFlow()
+                .getJavadocGeneration()
+                .getStrategy() == JavadocGenerationStrategy.IGNORE)) {
```
Comment (thescouser89) [inline pig/src/main/java/org/jboss/pnc/bacon/pig/PigFacade.java:197]: Should that be `!= IGNORE`?

File: pig/src/main/java/org/jboss/pnc/bacon/pig/PigFacade.java [hunk 5]
```diff
@@ -162,28 +166,44 @@ public static GroupBuildInfo run(
...
             generateJavadoc();
         } else {
             log.info("Skipping Javadoc Generation");
         }
 
-        if (!skipLicenses && repo != null) {
+        if (!(skipLicenses || context.getPigConfiguration()
+                .getFlow()
+                .getLicensesGeneration()
+                .getStrategy() == LicenseGenerationStrategy.IGNORE)) {
```
Comment (thescouser89) [inline pig/src/main/java/org/jboss/pnc/bacon/pig/PigFacade.java:206]: Should that be `!= IGNORE`?

File: pig/src/main/java/org/jboss/pnc/bacon/pig/PigFacade.java (hunk 4 above)
Comment (jomrazek) [reply pig/src/main/java/org/jboss/pnc/bacon/pig/PigFacade.java:197]: I think I have it ok, `(if **skip**=true OR **strategy**=true)`**!**(don't do if do else)

Comment (thescouser89) [reply pig/src/main/java/org/jboss/pnc/bacon/pig/PigFacade.java:197]: oh right! I didn't notice the `!` at the begininning.

**Pull request merged**